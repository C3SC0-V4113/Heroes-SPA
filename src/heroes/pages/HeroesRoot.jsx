import { Outlet } from "react-router-dom";
import { NavBar } from "../components";

export const HeroesRoot = () => {
  return (
    <>
      <NavBar />
      <div className="container">
        <Outlet />
      </div>
    </>
  );
};
