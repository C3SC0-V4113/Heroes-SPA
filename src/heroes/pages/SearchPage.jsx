// import { HeroCard } from "../components/HeroCard";

import { useLocation, useNavigate } from "react-router-dom";
import { useForm } from "../../hooks/useForm";
import queryString from "query-string";

export const SearchPage = () => {
  const navigate = useNavigate();
  const location = useLocation();
  // console.log(location);
  const { q = "" } = queryString.parse(location.search);

  const { searchText, onInputChange } = useForm({
    searchText: "",
  });

  const onSearchSubmit = (event) => {
    event.preventDefault();
    if (searchText.trim().length <= 1) return;
    navigate(`?q=${searchText.toLowerCase()}`);
  };

  return (
    <>
      <h1>Search</h1>
      <hr />
      <div className="row">
        <div className="col-5">
          <h4>Searching</h4>
          <hr />
          <form onSubmit={onSearchSubmit}>
            <input
              type="text"
              placeholder="Search a hero"
              className="form-control"
              name="searchText"
              autoComplete="off"
              value={searchText}
              onChange={onInputChange}
            />
            <button className="btn btn-outline-primary mt-3">Search</button>
          </form>
        </div>
        <div className="col-7">
          <h4>Results</h4>
          <hr />
          <div className="alert alert-primary">Search a Hero</div>
          <div className="alert alert-danger">
            Hero <b>{q}</b> not found
          </div>
          {/* <HeroCard /> */}
        </div>
      </div>
    </>
  );
};
