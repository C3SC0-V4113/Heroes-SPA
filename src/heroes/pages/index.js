export * from "./DcPage";
export * from "./MarvelPage";
export * from "./HeroesRoot";
