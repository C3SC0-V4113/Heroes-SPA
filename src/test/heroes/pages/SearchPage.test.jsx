import { render, screen } from "@testing-library/react";
import { SearchPage } from "../../../heroes/pages/SearchPage";
import { MemoryRouter } from "react-router-dom";

describe("pruebas en <SearchPage/>", () => {
  test("debe de mostrarse correctamente con valores por defecto", () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();

    // screen.debug();
  });

  test("debe de mostrar a batman y el input con el valor de queryString", () => {
    render(
      <MemoryRouter initialEntries={["/search?q=batman"]}>
        <SearchPage />
      </MemoryRouter>
    );

    const input = screen.getByRole("textbox");
    expect(input.value).toBe("batman");

    const img = screen.getByRole("img");
    expect(img.src).toContain("/heroes/dc-batman.jpg");

    const divSearch = screen.getByLabelText("div_search");
    expect(divSearch.style.display).toBe("none");
  });
});
