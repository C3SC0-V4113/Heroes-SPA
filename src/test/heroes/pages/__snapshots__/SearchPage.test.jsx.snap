// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <SearchPage/> debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-3"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        aria-label="div_search"
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a Hero
      </div>
      <div
        aria-label="div_not-found"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        Hero 
        <b />
         not found
      </div>
    </div>
  </div>
</div>
`;
